exonum-java-binding-core/src/main/java/com/exonum/binding/storage/database/IncrementalModificationCounter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.storage.database;
+
+final class IncrementalModificationCounter implements ModificationCounter {
+
+  private static final int INITIAL_VALUE = 0;
```

Comment:
Seems excessive to me, as we only use it once in the next line. Or maybe make it public? That way it'd be clear for users what the initial value is (because right now in tests to get `initialValue` we call `getCurrentValue` which is error-prone. We could use a public field in tests as well). But I do not insist, it's good as it is as well :)

Reply:
I do not think that the clients _shall_ care about the initial value, what they shall is _current_ value (= the value at the moment when an (spl)iterator is bound to source). _initial_ value is an implementation detail of this counter.

The constnant can probably be inlined, and the field renamed to `counter`